mongorestore/restore.go

```diff
@@ -372,8 +372,9 @@ func (restore *MongoRestore) RestoreCollectionToDB(dbName, colName string,
 			if doc == nil {
 				break
 			}
-			select {
-			case <-restore.termChan:
+
+			switch restore.terminate {
+			case true:
```

Comment:
[nit] might as well change this from a `switch` statement at this point, since we're checking only a single boolean value for `true`:

```
if restore.terminate {
    log.Logvf(...)
    termErr = ...
    close(...)
    return
}

rawBytes := ...
...
```